lib/chef/provider/package/dpkg.rb

```diff
@@ -119,22 +97,65 @@ def upgrade_package(name, version)
         end
 
         def preseed_package(preseed_file)
-          Chef::Log.info("#{@new_resource} pre-seeding package installation instructions")
+          Chef::Log.info("#{new_resource} pre-seeding package installation instructions")
           run_noninteractive("debconf-set-selections #{preseed_file}")
         end
 
         def reconfig_package(name, version)
-          Chef::Log.info("#{@new_resource} reconfiguring")
+          Chef::Log.info("#{new_resource} reconfiguring")
           run_noninteractive("dpkg-reconfigure #{name}")
         end
 
+        private
+
+        def get_current_version
+          Chef::Log.debug("#{new_resource} checking install state")
+          status = shell_out_with_timeout("dpkg -s #{current_resource.package_name}")
+          package_installed = false
+          status.stdout.each_line do |line|
+            case line
+            when DPKG_INSTALLED
+              package_installed = true
+            when DPKG_VERSION
+              if package_installed
+                Chef::Log.debug("#{new_resource} current version is #{$1}")
+                return $1
+              end
+            end
+          end
+          return nil
+        end
+
         # Runs command via shell_out_with_timeout with magic environment to disable
         # interactive prompts. Command is run with default localization rather
         # than forcing locale to "C", so command output may not be stable.
-        #
-        # FIXME: This should be "LC_ALL" => "en_US.UTF-8" in order to stabilize the output and get UTF-8
         def run_noninteractive(command)
-          shell_out_with_timeout!(command, :env => { "DEBIAN_FRONTEND" => "noninteractive", "LC_ALL" => nil })
+          shell_out_with_timeout!(command, :env => { "DEBIAN_FRONTEND" => "noninteractive" })
```

Comment:
At some point we also tore it out of all the rest of the package providers, including the apt provider so that now they call shell_out_with_timeout! which calls shell_out! and this is the only one that still overrides its LC_ALL setting.  Pretty certain this line of code is just a cargo cult now.  If it wasn't then the changes introduced in the apt provider would have broken the world a long time ago.